pkg/scheduler/profile/profile_test.go

```diff
@@ -251,6 +251,14 @@ func TestNewMap(t *testing.T) {
 			ctx, cancel := context.WithCancel(ctx)
 			defer cancel()
 			m, err := NewMap(ctx, tc.cfgs, fakeRegistry, nilRecorderFactory)
```

Comment:
looks like we should check `err` from NewMap?

Reply:
Ah, `checkErr` below does that.
Then, let's move defer after `checkErr`?

Reply:
My point is, when `err != nil`, it's likely `m` is nil and `defer` would panic.

Reply:
I will revise it and add some test cases

Reply:
if `defer` func, i add `if m != nil `